File: dashboard/src/app/@shared/models/payment-plan.ts
import { IEntity } from './entity';

export class PaymentPlan implements IEntity {
  public id: number;
  public firstAmount: number;
  public recurringAmount: number;
  public recurringMonths: number;
  public customerId: number;
  public blendId: number;
  public externalReference: string;
  public creationDate: Date;
  public lastModified: Date;
  public status: string;
  public paymentDate: Date;
  public stopReason: string;
  public deliveryDate: Date;
  public nextPaymentDate: Date;
  public nextDeliveryDate: Date;
  public translatedStatus?: string;
  public couponCode?: string;
  public entityName = 'PaymentPlan';
  public constructor(init?: Partial<PaymentPlan>) {
    Object.assign(this, init);
  }
}

File: backend/src/main/java/viteezy/domain/payment/PaymentPlan.java
package viteezy.domain.payment;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.Objects;
import java.util.Optional;
import java.util.StringJoiner;
import java.util.UUID;

public record PaymentPlan(Long id, BigDecimal firstAmount, BigDecimal recurringAmount, Integer recurringMonths,
                          Long customerId, Long blendId, UUID externalReference, LocalDateTime creationDate,
                          LocalDateTime lastModified, PaymentPlanStatus status, LocalDateTime paymentDate,
                          String stopReason, LocalDateTime deliveryDate, Optional<LocalDateTime> nextPaymentDate,
                          Optional<LocalDateTime> nextDeliveryDate, String paymentMethod) {
}
